src/main/kotlin/app/revanced/util/fingerprint/MethodWithReferenceInstructionFingerprint.kt

```diff
@@ -0,0 +1,30 @@
+package app.revanced.util.fingerprint
+
+import app.revanced.extensions.InstructionExtensions
+import app.revanced.extensions.InstructionExtensions.referenceEquals
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import app.revanced.util.fingerprint.FingerprintUtils.hasInstruction
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.reference.MethodReference
+import java.util.*
+
+abstract class MethodWithReferenceInstructionFingerprint(
+    internal val returnType: String? = null,
+    internal val accessFlags: Int? = null,
+    internal val parameters: Iterable<String>? = null,
+    internal val opcodes: Iterable<Opcode?>? = null,
+    internal val strings: Iterable<String>? = null,
+    instructionOpcode: Opcode,
+    methodReference: MethodReference,
+    matchSettings: EnumSet<InstructionExtensions.MethodReferenceMatch> = InstructionExtensions.MethodReferenceMatch.all
```

Comment:
Any reason these are not internal?

Reply:
Parameters can only be `internal` if they are also marked as `val`, but we don't need them as value because they are never used outside of the superclass invokation; But if you want I can make them `internal val`